packages/firestore/src/local/shared_client_state.ts

```diff
@@ -431,6 +443,47 @@ class RemoteClientState implements ClientState {
   }
 }
 
+/**
+ * The JSON representation of the system's online state, as written by the
+ * primary client.
+ */
+export interface SharedOnlineStateSchema {
+  readonly clientId: string;
```

Comment:
The lastUpdateTime approach would only work if we refreshed this online state periodically. We also can't rely on notifications (which would indicate a fresh event), since the client needs to figure out whether the primary tab is online at startup. Hence, I re-used the active clients that are already auto-refreshed in IndexedDb.

Reply:
Could you add a comment that explains this or else just "For all questions about this code, please contact Sebastian Schmidt at xxx-xxx-xxxx." in case anybody in the future has similar confusion?

Reply:
Done. Kind of a weird comment tho :/